### PR TITLE
[SPARK-27029][BUILD] Update Thrift to 0.12.0

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Update Thrift to 0.12.0 to pick up bug and security fixes.
Changes: https://github.com/apache/thrift/blob/master/CHANGES.md
The important one is for https://issues.apache.org/jira/browse/THRIFT-4506

## How was this patch tested?

Existing tests. A quick local test suggests this works.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1945,7 +1944,7 @@
       <dependency>
         <groupId>org.apache.thrift</groupId>
         <artifactId>libfb303</artifactId>
-        <version>${libthrift.version}</version>
+        <version>0.9.3</version>
```
Comment (felixcheung) [inline pom.xml]: it's ok this is 0.9.3?

**Review by felixcheung — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (felixcheung) [reply pom.xml]: I mean mixin with libthrift 0.12

Comment (srowen) [reply pom.xml]: Yeah, there's no newer version of libfb303. It seems to work though.

File: pom.xml [hunk 2]
```diff
@@ -179,7 +179,6 @@
     <joda.version>2.9.3</joda.version>
     <jodd.version>3.5.2</jodd.version>
     <jsr305.version>3.0.0</jsr305.version>
-    <libthrift.version>0.9.3</libthrift.version>
```
Comment (dongjoon-hyun) [inline pom.xml]: @srowen . Can we keep this? For example, `jsr305.version` (at above line) is also used once but we have it.

Comment (srowen) [reply pom.xml]: Sure that's fine. Normally I'd say it's just complicating the build but it also gives others a hook to change this version without maintaining a modification to the build file

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM, too. Merged to master according to the target version on JIRA.
Thank you for this security fix, @srowen.

**Pull request closed without merging**